### ISSUE TITLE
Students retaking completed quizzes

### ISSUE THREAD

**Issue by stianr** [cul-2016/quiz#473]
As a student, I want be able to have another go at the quizzes that have already been run.

Current functionality is that students can review the previous quizzes, and see (a) the questions and options, (b) their response, (c) the correct answer if different. All questions in a given quiz currently appear in a single page.

New functionality would be to present the quiz again - possibly using a similar format to that of the live quiz, but no strong view if a different format would be better. A good flow would be 

1. Student clicks on a particular quiz on their module overview page.

1. Interstitial page appears with the title of the quiz, a button "Try Quiz Again", text saying "This is just for revision - it won't change the score you got for the quiz", and a Back button. Pressing Back returns to the module overview page they came from, pressing Try Quiz Again starts the quiz.

1. First question is presented with clickable options

1. On first click of one of the options, prevent any further selections and check whether option is correct. If correct, highlight the selected option in green. If incorrect, highlight in red, and show correct option in green. Display a 'Next' button if there are further questions to come. 

1. At the end of the last question have a page with text "Quiz Complete!" and "Return to module" button, which returns the user to the module overview page.

Ideally the quiz would keep track of the user's score and present it at the end as feedback. 

Re styling, I'd thought of presenting questions individually as in the live quiz, but if it's easier (given the current formatting) we could have the questions presented on a single page, and users scroll down, as they do in Buzzfeed quizzes like [this](https://www.buzzfeed.com/eeerin/if-you-can-get-99-on-this-capital-quiz-you-are-p-3407t).

Things to bear in mind

- We may later want to present lecturer-generated feedback about the question, which may be worth thinking about now re design.
- Further down the line we may want to save scores at the end of the quiz

**Comment by sohilpandya** [comment]
@jackcarlisle & I feel that it's nice to have a place to look at your original quiz with answers like we have currently and then implement the try quiz button on the module dashboard next to each of the quizzes. 

Happy to discuss this point at the meeting today.

**Comment by stianr** [comment]
I like this - so if students click on the medal, they'll see their answers, as currently, but if they click on the new button, they'll be able to try it as a quiz. Best of both worlds, really.

**Comment by stianr** [comment]
As discussed, we looked at a compromise to cut down development time. This would:

- Load all questions and options into a single page, as the current review-quiz page does

- Hide the information about which answer is correct, and display each question with its options, and a 'Show Answer' button below

- On pressing the button, reveal the correct answer for that question

**Comment by sohilpandya** [comment]
potential steps for the above feature:
- create a new set of buttons
    - revise
    - review
- Revise will take the user to the new screen that we will build.
- The review will take the user to the existing quiz review screen. 
- Revise screen should look similar to review screen with the exception of a `show answer` button. 
- each button will be connected to a given question and selecting it will reveal the options the user chose during the live quiz.

**Comment by stianr** [comment]
This sounds very sensible. One question would be how the flow should go. At the moment, students click on the medal for a particular quiz, and then see the quiz with correct answer/chosen answer. I think there are two options. One would be to have two buttons next to each medal - 'Revise' and 'Review'. That would then be clear, but a little cluttered. An alternative would be to have a single press on the medal open up an interstitial page or within-page popup with 'Do you want to' and 'Revise', 'Review' and 'Back' buttons. That would be cleaner, but possibly more work, and might be less clear to students what their options are. Think I slightly prefer the second option, but if would take much longer, happy to go with the first.

**Comment by sohilpandya** [comment]
@stianr yeah, that sounds good, we'll probably tackle it the first way as the designs for this page are likely to change. so we would rather leave that extra time for the redesign instead of doing more work initially that might have to change.

**Comment by iteles** [comment]
Removed from sprint milestone as agreed in sprint planning. Time preference given to #518

**Comment by iteles** [comment]
The revise portion of this was implemented @stianr ! We may want to leave the issue open and remove the milestone so future work can be carried out to further this feature if you deem it useful?

**Comment by stianr** [comment]
That makes sense - current state is great, but would be good in future to allow students to choose an option, so will leave open. Removing please-test, as it all seems to be working well.
